### ISSUE TITLE
Change in CoreBooted prevented server from starting, needs explanation

### ISSUE THREAD

**Issue by benjamingeer** [dhlab-basel/Knora#46]
One of the changes in pull request #35 prevented `re-start` from starting the API server:

```
-    implicit lazy val system = ActorSystem("webapi")
+    implicit def system = ActorSystem("webapi")
```

The symptom was that the server would appear to start (the console messages looked normal), but any HTTP request from a browser would time out. The user would just get the message `The server was not able to produce a timely response to your request` in the browser, and there would be no additional output on the console.

I changed it back to `lazy val`, and now `re-start` works again. I don't understand why this needs to be a `lazy val`, or why it was changed to a `def`. This needs a comment in the code to explain it, so the same problem doesn't happen again.

Also, the Scaladoc comments for `Core` and `CoreBooted` are not very helpful. They say that these classes are part of the cake pattern, but it would be better to explain what the cake pattern is accomplishing here. Is the intention that we could make another class like `KnoraService`, but with a fake `ActorSystem`? If so, why would we want to do that?

**Comment by benjamingeer** [comment]
Also, we need a test that checks whether the server responds to HTTP requests when it's started using `Main`. For some reason, `ResourcesV1E2ESpec` didn't catch this problem.

**Comment by subotic** [comment]
Sorry, my bad. I don' really know why I changed this.

As far as the cake pattern goes, we used to need it together with the play framework for testing. Now we don't actually need it anymore, as I think that we can test everything without it. At least none of our current tests utilize it. Maybe the easiest would be to refactor it out and make the startup simpler?

**Comment by benjamingeer** [comment]
OK! Could you please make a pull request for that?

**Comment by subotic** [comment]
Fixed in #47
